src/cascadia/TerminalControl/ControlCore.h

```diff
@@ -50,6 +50,31 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         WINRT_PROPERTY(bool, IsIndex16);
     };
 
+    struct SearchState
+    {
+    public:
+        static std::atomic<size_t> _searchIdGenerator;
+
+        SearchState(const winrt::hstring& text, const Search::Sensitivity sensitivity, const bool forward) :
+            text(text),
+            sensitivity(sensitivity),
+            goForward(forward),
+            searchId(_searchIdGenerator.fetch_add(1))
+        {
+        }
+
+        std::optional<std::vector<std::pair<til::point, til::point>>> matches;

```

Comment:
Not sure it would ultimately make that much of a difference. Like, 4B per match (24B vs 32B) while the search runs? seems... meh to me. Plus, search already operates on a pair of points. Might be prohibitively annoying to try and store as a start and length